app/src/main/res/layout/fragment_file_view.xml

```diff
@@ -36,18 +36,56 @@
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintTop_toTopOf="parent"
             android:background="@color/black"
-            android:visibility="@{viewModel.isPDFViewVisible ? View.VISIBLE : View.GONE}" />
+            android:visibility="@{viewModel.isPDF ? View.VISIBLE : View.GONE}" />
 
         <androidx.core.widget.ContentLoadingProgressBar
             android:id="@+id/progressBar"
             style="@style/Widget.AppCompat.ProgressBar"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
+            android:layout_marginBottom="68dp"
             android:indeterminate="true"
             android:visibility="@{viewModel.isBusy ? View.VISIBLE : View.GONE}"
             app:layout_constraintBottom_toBottomOf="parent"
             app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/fileViewError"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/generic_error"
+            android:textColor="@color/white"
+            android:textSize="16sp"
+            android:visibility="@{viewModel.isError? View.VISIBLE : View.GONE}"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/progressBar"/>
+
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/btnCancelAll"
+            style="@style/customButtonBottomSheetStyle"
+            android:layout_width="87dp"
```

Comment:
We shouldn't hardcode button widths as with larger texts it will be cut out. The same thing can happen when translations are used since some have more characters than others.

Also if you used "dp" make sure they can be split into equal parts. Instead of 87 you would use 88. This is because devices have 200% pixel densities, or 75% and it will look better if the number is not using odd numbers.